src/java.base/share/classes/sun/security/provider/HSS.java

```diff
@@ -0,0 +1,765 @@
+                }
+                throw new InvalidKeySpecException("keySpec is not an X509 one");
+            }
+            throw new InvalidKeySpecException("Wrong key format or key algorithm");
+        }
+
+        @Override
+        protected Key engineTranslateKey(Key key) throws InvalidKeyException {
+            if (key == null) {
+                throw  new InvalidKeyException("key cannot be null");
+            }
+            PublicKey pKey;
+            try {
+                // Check if key originates from this factory
+                if (key instanceof HSSPublicKey) {
+                    return key;
+                }
+                // Convert key to spec
+                X509EncodedKeySpec x509EncodedKeySpec
+                        = engineGetKeySpec(key, X509EncodedKeySpec.class);
+                // Create key from spec, and return it
+                pKey = engineGeneratePublic(x509EncodedKeySpec);
+            } catch (InvalidKeySpecException e) {
+                throw new InvalidKeyException(e);
+            }
+            return pKey;
+        }
+    }
+
+    static class HSSPublicKey extends X509Key implements Serializable, Length {
+        @Serial
+        private static final long serialVersionUID = 21;
+        private int L;
+        private transient LMSPublicKey lmsPublicKey;
+
+        HSSPublicKey(byte[] keyArray) throws InvalidKeyException {
+            int inLen = keyArray.length;
+            if (inLen < 4) {
+                throw new InvalidKeyException("HSS public key too short");
+            }
+            L = LMSUtils.fourBytesToInt(keyArray, 0);
+            lmsPublicKey = new LMSPublicKey(Arrays.copyOfRange(keyArray, 4, keyArray.length), 0, true);
+            algid = new AlgorithmId(ObjectIdentifier.of(KnownOIDs.HSSLMS));
+            byte[] derEncodedKeyarray = new DerOutputStream().putOctetString(keyArray).toByteArray();
+            this.setKey(new BitArray(8 * derEncodedKeyarray.length, derEncodedKeyarray));
+        }
+
+        @Override
+        public String toString() {
+            HexDumpEncoder  encoder = new HexDumpEncoder();
+
+            return "HSS/LMS public key, number of layers: " + L +
+                    ", LMS type: " + LMSUtils.lmsType(lmsPublicKey.type) +
+                    ",\nOTS type: " + LMSUtils.lmotsType(lmsPublicKey.otsType) +
+                    ", byte array representation:\n" +
+                    encoder.encode(getKey().toByteArray());
+        }
+
+        /**
+         * Parse the key. Called by X509Key.
+         */
+        @Override
+        protected void parseKeyBits() throws InvalidKeyException {
+            byte[] keyArray = getKey().toByteArray();
+            L = LMSUtils.fourBytesToInt(keyArray, 0);
+            lmsPublicKey = new LMSPublicKey(keyArray, 0, true);
+        }
+
+
+        @java.io.Serial
+        protected Object writeReplace() throws java.io.ObjectStreamException {
```

Comment:
I think the serialized form of an HSSPublicKey should also be specified in the CSR since this Key is returned from a standard API. I think you can add a simple sentence such as:

"The Keys returned by an "HSS/LMS" `KeyFactory` are `Serializable` and use `java.security.KeyRep` as its serialized representation."